### DISCUSSION TITLE
Device-Verfügbarkeit prüfen

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#498] [Q&A]
Ich wollte mir ein HMIP-PSM (u.a.) hinter meinen Gefrierschrank setzen um dort mal die HA-Stromverbrauchsstatistik zu ergänzen.

Nur möchte ich auch gern absichern, dass ich mir einen Alarm sende, wenn das Ding mal ausfällt. 

Gibt es da Best Practices? Aktuell sendet HA und CCU ja erst mal ewig, dass das Gerät noch da ist, selbst wenn ich es aus der Steckdose ziehe. Auch wenn ich die Sendezyklen komplett runterdrehe. D.h. das ist dann nur Geräteseitig und die CCU erwartet dennoch nicht analog zu den Werten ein Update? Doof/unlogisch.

Derzeit toggle ich Channel 4 z.B. alle 30 Minuten. Und bekomme dann ja eine Systemmeldung und aber auch unavailable als Status der Entitäten, wenn das nicht ausgeführt werden kann, da Gerät weg. Auf ausbleibende Spannungsänderung, last updated zu prüfen, usw. kann ja auch mal eine Stunde oder länger dauern.

Aber ist das wirklich so kompliziert? Oder geht es einfacher oder besser? Ich hätte irgendwie gedacht/erwartet (HA und Integration können da ja wohl nichts für), dass die CCU irgendwie früher mitbekommt, wenn ein Gerät nicht mehr da ist.

**Comment by Baxxy13** [comment]
Bei HM wird relativ schnell erkannt wenn ein Gerät "weg" ist. Bei IP konnte ich bisher noch kein Muster erkennen.
Testweise würde ich mal die zykl. Statusmeldungen auf 0/0 setzen und dann das Teil entstromen.

Was kann man machen:

- WebUI-Programm was bei Aktualisierung von "irgendwas in Kanal:0" eine Alarm-SysVar verzögert um X setzt. X muss getestet werden, i.d.R. melden sich IP Geräte spätestens nach 90m
- Stromausfallüberwachung HmIP-PMFS zwischen PSM und Gefrierschrank. Ggf. auch Selbstbau mit Koppelrelais und Kontaktschnittstelle
- Temperaturüberwachung Innenraum Gefrierschrank bspw. mit HmIP-STE2-PCB oder anderen Sensoren

**Reply by emufan** [reply]
Das mit den Statusmeldungen hatte ich ja schon gemacht. Scheint den Aktor zu beeinflussen, aber die CCU erwartet dann dennoch nicht häufiger Rückmeldung. Hätte ich zumindest korrelierend gebaut. ;-)

Der Rest klingt zumindest bisher nicht schneller/einfacher als mein periodisches Sicherheits-Togglen eines nicht verwendeten Kanals. Denn dann geht das Gerät im Fehlerfall ja direkt auf nicht "unavailable" im HA.
